src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -355,7 +356,17 @@ private static int FindAndParseNextNumber(string text, ref int pos)
                     num = (num * 10) + (c - '0');
                 }
                 else break;
+
+                // Integer overflow can occur for example with:
+                //     Linux nelknet 4.15.0-24201807041620-generic
+                // To form a valid Version, num must be positive.
+                if (num < 0)
```

Comment:
I think the expression could have overflowed without `num` being negative here. Try text="10000000000", pos=0.

Two alternatives I see are:

1. Use `checked` and catch the OverflowException
2. Do 64-bit math and bail when int.MaxValue is exceeded.

Reply:
Again, you're right.